File: desafio-02/pedrobenicio2005/c/primos.c
#include <stdio.h>

int eh_primo(int numero) {
    if (numero <= 1) {
        return 0;
    }
    if (numero == 2) {
        return 1;
    }
    if (numero % 2 == 0) {
        return 0;
    }

    for (int i = 3; i * i <= numero; i += 2) {
        if (numero % i == 0) {
            return 0;
        }
    }
    return 1;
}

int main() {
    printf("Numeros primos entre 1 e 10000:\n\n");

    int contador = 0;

    for (int n = 1; n <= 10000; n++) {
        if (eh_primo(n)) {
            printf("%d ", n);
            contador++;

            if (contador % 10 == 0) {
                printf("\n");
            }
        }
    }

    printf("\n\nTotal de numeros primos: %d\n", contador);

    return 0;
}

File: desafio-02/doctor45dev/java/Primos.java
public class Primos {
  public static boolean ehPrimo(int numero) {
    if (numero <= 1) {
      return false;
    }

    for (int i = 2; i * i <= numero; i++) {
      if (numero % i == 0) {
        return false;
      }
    }
    return true;
  }

  public static void main(String[] args) {
    for (int i = 1; i <= 10000; i++) {
      if (ehPrimo(i)) {
        System.out.println(i);
      }
    }
  }
}
